### PR TITLE
Additional r packages

### PR DESCRIPTION
**Pull request by mcneish1**
Add r packages needed for https://github.com/ECP-CANDLE/Supervisor

### CODE REVIEW

File: var/spack/repos/builtin/packages/r-bbmisc/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    """Miscellaneous helper functions for and from B. Bischl and some other
+       guys, mainly for package development."""
+
+    homepage = "https://github.com/berndbischl/BBmisc"
+    url      = "https://cran.r-project.org/src/contrib/BBmisc_1.11.tar.gz"
+    list_url = "https://cran.r-project.org/src/contrib/Archive/BBmisc"
+
+    version('1.11', '681642628037406beb6088d5f773473d')
+
+    depends_on('r-checkmate', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-bbmisc/package.py]: `r-checkmate@1.8.0:`

File: var/spack/repos/builtin/packages/r-mlr/package.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    depends_on('r-paramhelpers', type=('build', 'run'))
+    depends_on('r-bbmisc', type=('build', 'run'))
+    depends_on('r-backports', type=('build', 'run'))
+    depends_on('r-ggplot2', type=('build', 'run'))
+    depends_on('r-stringi', type=('build', 'run'))
+    depends_on('r-checkmate', type=('build', 'run'))
+    depends_on('r-data-table', type=('build', 'run'))
+    depends_on('r-parallelmap', type=('build', 'run'))
+    depends_on('r-xml', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-mlr/package.py]: Can you add the specific version requirements?
https://cran.r-project.org/web/packages/mlr/index.html

File: var/spack/repos/builtin/packages/r-mlrmbo/package.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    depends_on('r-mlr', type=('build', 'run'))
+    depends_on('r-paramhelpers', type=('build', 'run'))
+    depends_on('r-smoof', type=('build', 'run'))
+    depends_on('r-backports', type=('build', 'run'))
+    depends_on('r-bbmisc', type=('build', 'run'))
+    depends_on('r-checkmate', type=('build', 'run'))
+    depends_on('r-data-table', type=('build', 'run'))
+    depends_on('r-lhs', type=('build', 'run'))
+    depends_on('r-parallelmap', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-mlrmbo/package.py]: Version requirements: https://cran.r-project.org/web/packages/mlrMBO/index.html

File: var/spack/repos/builtin/packages/r-parallelmap/package.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+       this operation."""
+
+    homepage = "https://github.com/berndbischl/parallelMap"
+    url      = "https://cran.r-project.org/src/contrib/parallelMap_1.3.tar.gz"
+    list_url = "https://cran.r-project.org/src/contrib/Archive/parallelMap"
+
+    version('1.3', 'dd62866b395847b0bd5b13bed98c0081')
+
+    depends_on('r-bbmisc', type=('build', 'run'))
+    depends_on('r-checkmate', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-parallelmap/package.py]: Version requirements: https://cran.r-project.org/web/packages/parallelMap/index.html

File: var/spack/repos/builtin/packages/r-paramhelpers/package.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+       function evaluations is also provided."""
+
+    homepage = "https://github.com/berndbischl/ParamHelpers"
+    url      = "https://cran.r-project.org/src/contrib/ParamHelpers_1.10.tar.gz"
+    list_url = "https://cran.r-project.org/src/contrib/Archive/ParamHelpers"
+
+    version('1.10', '36e9060488ebd484d62cd991a4693332')
+
+    depends_on('r-bbmisc', type=('build', 'run'))
+    depends_on('r-checkmate', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-paramhelpers/package.py]: Versions: https://cran.r-project.org/web/packages/ParamHelpers/index.html

File: var/spack/repos/builtin/packages/r-smoof/package.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    depends_on('r-bbmisc', type=('build', 'run'))
+    depends_on('r-checkmate', type=('build', 'run'))
+    depends_on('r-ggplot2', type=('build', 'run'))
+    depends_on('r-rcolorbrewer', type=('build', 'run'))
+    depends_on('r-plot3d', type=('build', 'run'))
+    depends_on('r-plotly', type=('build', 'run'))
+    depends_on('r-mco', type=('build', 'run'))
+    depends_on('r-rcpp', type=('build', 'run'))
+    depends_on('r-rjsonio', type=('build', 'run'))
+    depends_on('r-rcpparmadillo', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-smoof/package.py]: Versions: https://cran.r-project.org/web/packages/smoof/index.html

**Comment by mcneish1** [comment]
I'll go through and add specific requirements.

Do you know how I'm supposed to get codecov/project to pass?

**Comment by adamjstewart** [comment]
> Do you know how I'm supposed to get codecov/project to pass?

See the note at the bottom of: http://spack.readthedocs.io/en/latest/contribution_guide.html#coverage

**Comment by mcneish1** [comment]
Getting a `==> Error: Invalid Version range: 1.5.1:1.5` trying to install them all at once. Not sure where it's coming from yet.

File: var/spack/repos/builtin/packages/r-mco/package.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+##############################################################################
+from spack import *
+
+
+class RMco(RPackage):
+    """Functions for multiple criteria optimization using genetic algorithms
+       and related test problems"""
+
+    homepage = "https://github.com/cran/mco"
+    url      = "https://cran.r-project.org/src/contrib/Archive/mco/mco_1.0-15.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-mco/package.py]: Can you add the latest version:
https://cran.r-project.org/src/contrib/mco_1.0-15.1.tar.gz

File: var/spack/repos/builtin/packages/r-mlr/package.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+       example-specific cost-sensitive learning. Generic resampling,
+       including cross-validation, bootstrapping and subsampling.
+       Hyperparameter tuning with modern optimization techniques,
+       for single- and multi-objective problems. Filter and wrapper methods for
+       feature selection. Extension of basic learners with additional
+       operations common in machine learning, also allowing for easy nested
+       resampling. Most operations can be parallelized."""
+
+    homepage = "https://github.com/mlr-org/mlr/"
+    url      = "https://cran.r-project.org/src/contrib/Archive/mlr/mlr_2.12.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-mlr/package.py]: Can you add the latest version:
https://cran.r-project.org/src/contrib/mlr_2.12.1.tar.gz

File: var/spack/repos/builtin/packages/r-mlrmbo/package.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+       many different infill criteria to guide the search process. Additional
+       features include multi-point batch proposal, parallel execution as well
+       as visualization and sophisticated logging mechanisms, which is
+       especially useful for teaching and understanding of algorithm behavior.
+       'mlrMBO' is implemented in a modular fashion, such that single
+       components can be easily replaced or adapted by the user for specific
+       use cases."""
+
+    homepage = "https://github.com/mlr-org/mlrMBO/"
+    url      = "https://cran.r-project.org/src/contrib/Archive/mlrMBO/mlrMBO_1.1.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-mlrmbo/package.py]: Can you add the latest version:
https://cran.r-project.org/src/contrib/mlrMBO_1.1.1.tar.gz

File: var/spack/repos/builtin/packages/r-smoof/package.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+class RSmoof(RPackage):
+    """Provides generators for a high number of both single- and
+       multi- objective test functions which are frequently used for the
+       benchmarking of (numerical) optimization algorithms. Moreover, it offers
+       a set of convenient functions to generate, plot and work with objective
+       functions."""
+
+    homepage = "http://github.com/jakobbossek/smoof"
+    url      = "https://cran.r-project.org/src/contrib/Archive/smoof/smoof_1.5.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-smoof/package.py]: Can you add the latest version:
https://cran.r-project.org/src/contrib/smoof_1.5.1.tar.gz

This is causing the concretization error you are seeing.

**Comment by mcneish1** [comment]
Got that all fixed, testing it now then I'll push that commit.

On the concretization error,
Would it be possible to print the package causing that? I had to `spack spec` everything just to find the package that requested the impossible dependency, and in retrospect it's obvious that the 1.5.1 mlrMBO was requesting is r-smoof, but I didn't catch that my first time through.

**Comment by adamjstewart** [comment]
> Would it be possible to print the package causing that as part of the error message?

@alalazo would know more than I do on this one.

**Pull request merged**